### ISSUE TITLE
False-positive for functools.cache decorated methods

### ISSUE THREAD

**Issue by vemel** [microsoft/pyright#2578]
**Describe the bug**

`pyright` marks methods wrapped with `functools.cache`/`functools.lru_cache` decorator calls as `error: Argument missing for parameter "self"`. Functions are not affected, only methods.

This bug was introduced in `1.1.187`. `1.1.186` is not affected.

**To Reproduce**

Run `pyright` on this sample:

```python
from functools import cache


class A:
    @cache
    def has_cache(self) -> None:
        print(self)

    def no_cache(self) -> None:
        print(self)


a = A()
a.has_cache()  #  error: Argument missing for parameter "self"
a.has_cache(a)  # no typing errors, but obviously a RuntimeError
a.no_cache()  # all good
```

**Expected behavior**
No error

**VS Code extension or command-line**
Command line with `pyright 1.1.187`.

**Comment by erictraut** [comment]
Thanks for creating the issue. Based on my understanding of PEP 612, pyright's current behavior is correct.

I've asked for input from the PEP 612 authors and the broader Python typing community [here](https://github.com/python/typing/discussions/946).

**Comment by erictraut** [comment]
The typeshed maintainers have decided to revert the change to `functools.pyi` for now. I've incorporated the updated stubs, so this problem will be fixed in the next release of pyright.

**Comment by erictraut** [comment]
This is included in pyright 1.1.190, which I just published. It will also be included in the next release of pylance.
